modules/db/src/main/scala/gem/dao/SmartGcalDao.scala

```diff
@@ -0,0 +1,24 @@
+package gem.dao
+
+import gem.config.{F2SmartGcalKey, GcalConfig, SmartGcalKey}
+import gem.enum.{GcalBaselineType, GcalLampType}
+import doobie.imports._
+
+object SmartGcalDao {
+  def insert(l: GcalLampType, b: GcalBaselineType, k: SmartGcalKey, g: GcalConfig): ConnectionIO[Int] = {
+    def insertSmartF2(gcalId: Int, k: F2SmartGcalKey): ConnectionIO[Int] =
+      sql"""
+        INSERT INTO smart_f2 (lamp, baseline, disperser, filter, fpu, gcal_id)
+             VALUES ($l :: gcal_lamp_type, $b :: gcal_baseline_type, ${k.disperser}, ${k.filter}, ${k.fpu}, $gcalId)
```

Comment:
`gcal_lamp_type` is an `ENUM`:

````
CREATE TYPE gcal_lamp_type AS ENUM (
    'Arc',
    'Flat'
);
````

In `smart_f2` the `lamp` column is given type `gcal_lamp_type` but here I'm inserting text so a cast is needed.  I'm not sure if there's a way to work around this.  There seems to exist a `CREATE CAST` statement that might be used to make this implicit?